extensions/justablock/gitpenguin.js

```diff
@@ -0,0 +1,205 @@
+                {
+                    "opcode": "editFileContent",
+                    "blockType": Scratch.BlockType.COMMAND,
+                    "text": "edit content of file [FILE] in repository [REPO] of user [NAME] to [CONTENT] using token [TOKEN]",
+                    "arguments": {
+                        "FILE": {
+                            "type": Scratch.ArgumentType.STRING,
+                            "defaultValue": "fileToEdit.txt"
+                        },
+                        "REPO": {
+                            "type": Scratch.ArgumentType.STRING,
+                            "defaultValue": "repository"
+                        },
+                        "NAME": {
+                            "type": Scratch.ArgumentType.STRING,
+                            "defaultValue": "username"
+                        },
+                        "CONTENT": {
+                            "type": Scratch.ArgumentType.STRING,
+                            "defaultValue": "New content"
+                        },
+                        "TOKEN": {
+                            "type": Scratch.ArgumentType.STRING,
+                            "defaultValue": "YOUR_GITHUB_TOKEN"
+                        }
+                    }
+                },
+                {
+                    "blockType": Scratch.BlockType.LABEL,
+                    "text": "DANGEROUS"
+                },
+                {
+                    "opcode": "deleteFile",
+                    "blockType": Scratch.BlockType.COMMAND,
+                    "text": "delete file [FILE] from repository [REPO] of user [NAME] using token [TOKEN]",
+                    "arguments": {
+                        "FILE": {
+                            "type": Scratch.ArgumentType.STRING,
+                            "defaultValue": "fileToDelete.txt"
+                        },
+                        "REPO": {
+                            "type": Scratch.ArgumentType.STRING,
+                            "defaultValue": "repository"
+                        },
+                        "NAME": {
+                            "type": Scratch.ArgumentType.STRING,
+                            "defaultValue": "username"
+                        },
+                        "TOKEN": {
+                            "type": Scratch.ArgumentType.STRING,
+                            "defaultValue": "YOUR_GITHUB_TOKEN"
+                        }
+                    }
+                }
+            ]
+        };
+    }
+
+    async getFileContents({ FILE, REPO, NAME }) {
+        const apiUrl = `https://api.github.com/repos/${NAME}/${REPO}/contents/${FILE}`;
+        const response = await fetch(apiUrl);
+        const data = await response.json();
+        if (response.ok) {
+            const content = atob(data.content); // Decode base64 content
+            return content;
+        } else {
+            throw new Error(`Failed to fetch file: ${data.message}`);
```

Comment:
All reporters **must** return something, as ScratchBlocks (and consequentially, the vm) expect a value of String, a Number and Boolean.